### ISSUE TITLE
Bug: sometimes files are exported with wrong extension

### ISSUE THREAD

**Issue by RhetTbull** [RhetTbull/osxphotos#155]
Dear @RhetTbull I tested the --update option and works very well! But during the execution I receive some "WARNING" that I believe related to the export of some files. Even if I use the --verbose option in the log I don't find any errors. Is there any test that I can do to try to find out more precisely, what generates these errors? Thank you!
<img width="1333" alt="Schermata 2020-06-05 alle 11 58 03" src="https://user-images.githubusercontent.com/7814756/83921822-80b12680-a77f-11ea-85b3-5c65e8471b2e.png">

_Originally posted by @NiccoStar86 in https://github.com/RhetTbull/osxphotos/issues/100#issuecomment-639812627_

**Comment by RhetTbull** [comment]
@NiccoStar86 I've pushed a new version, 0.29.11.  This fixes the '.tif' to '.tiff' warnings and adds additional details to the warning for the other files.  If you re-run the same command that would be helpful.  Also, there's a new `--dry-run` option which causes `osxphotos` to not actually export any photos so you can try that with a new destination directory which will cause a re-export of your entire database.  But, since you used '--dry-run`, no actual photos will be copied though you'll see any warnings that are generated.

**Comment by NiccoStar86** [comment]
Hi @RhetTbull I tested again the export with the version 0.29.13 and I received only this warning:
<img width="1473" alt="Schermata 2020-06-08 alle 00 13 46" src="https://user-images.githubusercontent.com/7814756/83981335-35347f00-a91d-11ea-86f2-8ef9fefd72b6.png">

I searched for the file stored in the _originals_ folder and I found it, with .png extension:
<img width="773" alt="Schermata 2020-06-08 alle 00 14 15" src="https://user-images.githubusercontent.com/7814756/83981359-68770e00-a91d-11ea-9edf-55c42f2a5ace.png">

I searched for the _original_filename_ in Photos and I found it, with .JPG extension:
<img width="596" alt="Schermata 2020-06-08 alle 00 13 28" src="https://user-images.githubusercontent.com/7814756/83981380-8d6b8100-a91d-11ea-90c6-f0ab00388db9.png">

How is it possible that a photo with .JPG is linked to a file with .png?? :) ... I could try to backup the photo, delete it from Photos, and import again, then launch again the procedure, and check the result if this can help you. Maybe it's just a sporadic case, if we consider that I exported 44614 photos without problems ... Let me know :)

**Comment by RhetTbull** [comment]
@NiccoStar86 that's indeed strange -- I've never seen this.  Screenshots taken on iPhone are in PNG format and sometimes store a copy as JPEG but this looks like an imported photo.  If you don't mind, pull the latest version from GitHub or PyPI (0.29.13) and try the following hidden command to get some other debug information:

`osxphotos debug-dump --dump photos --uuid 25EAA77E-61D0-48D7-98CC-9BC998E9DD3E > debug.txt`

This will dump some debug info about that particular photo to debug.txt. You can either send me the file or post results here and I'll see if I can debug this.

Cheers!

**Comment by NiccoStar86** [comment]
Hi @RhetTbull attached the result of the command! Thank you
[debug.txt](https://github.com/RhetTbull/osxphotos/files/4750455/debug.txt)

**Comment by NiccoStar86** [comment]
@RhetTbull I saw that already in Photos there is something of strange ...

<img width="596" alt="83981380-8d6b8100-a91d-11ea-90c6-f0ab00388db9" src="https://user-images.githubusercontent.com/7814756/84117124-51294500-aa31-11ea-966e-0e229d2752c3.png">

If after a "quick" check you don't find an explanation, I can try to delete and import the image again and see the result. I don't want to waste your time on a single photo! :)

**Comment by RhetTbull** [comment]
@NiccoStar86 Thanks for the debug data -- it clearly shows that Photos thinks this is a PNG that was imported as IMG_4745.JPG just as your screenshot shows.  For example, the UTI is set to PNG: `'UTI': 'public.png'`  I suggest you try to export the image from Photos (select photo then File | Export Unmodified original for 1 Photo) then in the Terminal, do 

`file IMG_4754.JPG` (assuming that's what Photos names it) -- it will either say something like:

`IMG_4754.JPG: JPEG image data, Exif standard: [TIFF image data, little-endian, direntries=12, manufacturer=NIKON CORPORATION, model=NIKON D5300, orientation=upper-left, xresolution=188, yresolution=196, resolutionunit=2, software=Ver.1.00 , datetime=2019:12:03 18:45:01], baseline, precision 8, 6000x4000, components 3`

or 

`IMG_4754.JPG: PNG image data, 1536 x 2048, 8-bit/color RGB, non-interlaced`

If the latter, it's really a PNG image (just as osxphotos thought it was) but has a wrong name.

**Comment by NiccoStar86** [comment]
@RhetTbull I think something went wrong during the first import of the file, because by exporting the image as "original - unmodified" the saved file is ".png" ...
<img width="722" alt="Schermata 2020-06-09 alle 09 49 25" src="https://user-images.githubusercontent.com/7814756/84120897-bc294a80-aa36-11ea-9b39-9ee6954fd231.png">

**Comment by RhetTbull** [comment]
@NiccoStar86 I think you're right this was some sort of corrupted import.  I tried to replicate this for testing by renaming a PNG to jpeg then importing but I was not able to get the behavior you reported.  Photos added the image as a jpeg, set the UTI to jpeg, and when exported, named the photo .jpeg.  However, the image was clearly a PNG when inspected.  It seems the code in Photos that handles the database looks only at file extension and it is independent of the code that displays photos which is definitely reading the photo type from the actual file independent of the extension.  I'm going to close this issue but definitely let me know if you see more problems like this.

**Comment by NiccoStar86** [comment]
@RhetTbull I totally agree with you. I deleted and re-imported the exported photo in Photos, then I execute again the osxphotos with --update without any error!

<img width="1177" alt="Schermata 2020-06-10 alle 08 56 02" src="https://user-images.githubusercontent.com/7814756/84236796-8c427b80-aaf8-11ea-9914-e2f3c98c4f65.png">

... and the photo was correctly exported, with the right extension.

<img width="882" alt="Schermata 2020-06-10 alle 08 56 14" src="https://user-images.githubusercontent.com/7814756/84236837-9b292e00-aaf8-11ea-9020-3d8a557c36d4.png">

You can close this issue, and thank you!
